src/mpi-base.jl

```diff
@@ -156,6 +120,54 @@ function Comm_size(comm::Comm)
     Int(size[])
 end
 
+function Type_create_struct{T <: Any}(::Type{T})  # <: Any effectively 
```

Comment:
This function `Type_create_struct` has an API quite different from the underlying MPI function -- it commits the type, it doesn't return the new MPI type, etc.

I would thus give this function a new name, maybe simply `type_create`. (Following the Python convention, the functions with lower case names correspond to a higher level interface.)